src/Channels.Networking.Sockets/SocketConnection.cs

```diff
@@ -109,6 +109,10 @@ protected virtual void Dispose(bool disposing)
         {
             if (disposing)
             {
+                _input?.CompleteReading();
```

Comment:
Dispose needs to look over. It's interesting because there's CompleteReading() and CompleteWriting() but there's no way to wait until everything has unwound gracefully.